### PR TITLE
Passthrough arch flag for Simulator

### PR DESCRIPTION
**Pull request by lpusok**
### Context

If arch flag is specified, add it to the destination device. This allows explicitly requiring Rosetta Simulator.

https://bitrise.atlassian.net/browse/BE-1124

### CODE REVIEW

File: destination/device_finder_test.go [hunk 1]
```diff
@@ -90,6 +90,22 @@ func Test_deviceFinder_FindDevice(t *testing.T) {
 				OS:     "16.0",
 			},
 		},
+		{
+			name: "arch flag specified (rosetta Simulator)",
```
Comment (ofalvai) [inline destination/device_finder_test.go:94]:
```suggestion
			name: "arch flag specified (Rosetta Simulator)",
```

File: destination/simulator_test.go [hunk 2]
```diff
@@ -8,14 +8,14 @@ import (
 
 func Test_GivenDestinationIsCorrect_WhenSimulatorIsCreated_ThenItReturnsCorrectSimulator(t *testing.T) {
 	// Given
-	destination := "platform=iOS Simulator,name=iPhone 8 Plus,OS=latest"
+	destination := "platform=iOS Simulator,name=iPhone 8 Plus,OS=latest,arch=x86_64"
```
Comment (ofalvai) [inline destination/simulator_test.go:11]: I'd rather create a new test case for the arch flag to keep testing the no-arch-flag case as well.

File: destination/device_finder.go [hunk 3]
```diff
@@ -83,3 +85,12 @@ func (d deviceFinder) FindDevice(destination Simulator) (Device, error) {
...
 	return device, err
 }
+
+// XcodebuildDestination returns the required xcodebuild -destination flag value for a device
+func (d Device) XcodebuildDestination() string {
+	if d.Arch == "" {
+		return fmt.Sprintf("id=%s", d.ID)
+	}
+
+	return fmt.Sprintf("id=%s,arch=%s", d.ID, d.Arch)
```
Comment (ofalvai) [inline destination/device_finder.go:95]: Does this work? I thought we can't use the `id` at all with `arch` and need to use the rest of the arguments (`platform`, `os`, `device`).

**Comment by ofalvai** [comment]
Let's integrate this PR branch into the xcode-test step and run all tests before merging this

File: destination/simulator_test.go (hunk 2 above)
Comment (lpusok) [reply destination/simulator_test.go:25]: added new test

File: destination/device_finder.go [hunk 4]
```diff
@@ -83,3 +87,12 @@ func (d deviceFinder) FindDevice(destination Simulator) (Device, error) {
 
 	return device, err
 }
+
+// XcodebuildDestination returns the required xcodebuild -destination flag value for a device
+func (d Device) XcodebuildDestination() string {
+	if d.Arch == "" {
```
Comment (ofalvai) [inline destination/device_finder.go:93]:
```suggestion
	// `arch` doesn't seem to work together with `id`
	if d.Arch == "" {
```

**Review by ofalvai — APPROVED** [review]

**Pull request merged**